docs/articles/documentation/guides/advanced-guides/multiple-browser-windows.md

```diff
@@ -0,0 +1,138 @@
+
+    const url = await t.eval(() => document.documentURI);
+    await t.expect(url).eql('http://example.com/');
+});
+```
+
+>All open browser windows share a single pool of client-side storage. Only one [user role](https://devexpress.github.io/testcafe/documentation/guides/advanced-guides/authentication.html#user-roles) can be active at a time.
+
+## Switch between windows
+
+The [t.switchToWindow](../../reference/test-api/testcontroller/switchtowindow.md) method lets you switch between browser windows.
+
+It takes a window descriptor as an argument:
+
+```js
+fixture `Example page`
+    .page('http://example.com');
+
+test('Switch to a specific window', async t => {
+    const initialWindow = await t.getCurrentWindow();
+    const window2 = await t.openWindow('http://devexpress.com');
+    const window3 = await t.openWindow('http://github.com');
+
+    await t.switchToWindow(initialWindow);
+
+    const url = await t.eval(() => document.documentURI);
+    await t.expect(url).eql('http://example.com/');
+});
+```
+
+You can also pass [a predicate](../../reference/test-api/testcontroller/switchtowindow.md#tswitchtowindowpredicate) with the description of the window you need:
+
+```js
+await t.switchToWindow(w => w.title === 'Example Domain' && w.url.host === 'example.com');
+```
+
+Use the [t.switchToParentWindow](../../reference/test-api/testcontroller/switchtoparentwindow.md) method to access the window's parent:
+
+```js
+fixture `Example page`
+    .page('http://www.example.com/');
+
+test('Switch to a parent window', async t => {
+    await t
+        .openWindow('https://devexpress.com')
+        .switchToParentWindow();
+
+    const url = await t.eval(() => document.documentURI);
+    await t.expect(url).eql('http://www.example.com/');
+});
+```
+
+Use the [t.switchToPreviousWindow](../../reference/test-api/testcontroller/switchtopreviouswindow.md) method to access the second to last open window:
+
+```js
+test('Switch back', async t => {
+    await t
+        .openWindow('http://www.example.com')
+        .openWindow('https://devexpress.com');
+        .switchToPreviousWindow();
+
+    const url = await t.eval(() => document.documentURI);
+    await t.expect(url).eql('http://www.example.com/');
+});
+```
+
+## Close an existing window
+
+>You cannot close windows with open children.
+
+Call the [t.closeWindow](../../reference/test-api/testcontroller/closewindow.md) method without parameters to close the currently active window:
+
+```js
+fixture `Example page`
+    .page('http://www.example.com');
+
+test('Close the current window', async t => {
+    const window1 = await t.openWindow('http://devexpress.com');
+    await t.closeWindow();
+
+    const url = await t.eval(() => document.documentURI);
+    await t.expect(url).eql('http://www.example.com/');
+    });
+
+```
+
+Pass a window descriptor when you call the [t.closeWindow](../../reference/test-api/testcontroller/closewindow.md) method to close a particular window:
```

Comment:
```suggestion
Pass a window descriptor when you call the [t.closeWindow](../../reference/test-api/testcontroller/closewindow.md) method to close a window:
```

Reply:
I was trying to highlight the difference between what the t.closeWindow method does when called without arguments (it closes the currently active window) and what it does with a window descriptor present. Would `Pass a window descriptor when you call the [t.closeWindow](../../reference/test-api/testcontroller/closewindow.md) to close the corresponding window` work?